pandas/core/reshape/pivot.py

```diff
@@ -683,3 +697,35 @@ def _get_names(arrs, names, prefix="row"):
             names = list(names)
 
     return names
+
+
+def _build_names_mapper(names, shared_col_row_names, suffix):
+    keys, counts = _value_counts_arraylike(names, dropna=False)
```

Comment:
does using the non-private `factorize` not work here?

Reply:
Not sure is possible with `factorize`. The labels it outputs are not unique and we need the counts for deduplication.